modules/ingester/instance_search.go

```diff
@@ -417,8 +421,27 @@ func (i *instance) SearchTagValuesV2(ctx context.Context, req *tempopb.SearchTag
 	}
 
 	query := traceql.ExtractMatchers(req.Query)
+	// cacheKey will be same for all blocks in a request so only compute it once
+	cacheKey := searchTagValuesV2CacheKey(req, limit, "cache_search_tagvaluesv2")
```

Comment:
Good catch.  Do you know if we honor the time range here in the ingesters?  For example is there a difference when you autocomplete for `last 5 minutes` versus `last 30 minutes` ?  If so we might need account for that here.

The ideal case is:   we don't honor the time range once it hits the ingesters, and we zero out `req.Start` `req.End` before hashing, so that all incoming requests will match the existing cache, even if the time range was specified.

Reply:
we are passing start and end, and we adjust them to respect the `QueryIngestersUntil`

looks like we are not using start and end right now when handling the SearchTagValuesV2 but we have an TODO to make use of start and end so in future we might add support for start and end in ExecuteTagValues. https://github.com/grafana/tempo/blob/9fdc0324d8572366a0e6e40df728cb4416863f61/pkg/traceql/storage.go#L230

Reply:
done, dropped start and end from the cachekey and added a note.